core/src/main/java/org/apache/calcite/sql2rel/RelDecorrelator.java

```diff
@@ -476,14 +477,29 @@ protected RexNode removeCorrelationExpr(
 
     RelCollation oldCollation = rel.getCollation();
     RelCollation newCollation = RexUtil.apply(mapping, oldCollation);
-
-    final int offset = rel.offset == null ? -1 : RexLiteral.intValue(rel.offset);
-    final int fetch = rel.fetch == null ? -1 : RexLiteral.intValue(rel.fetch);
+    int offset = -1;
```

Comment:
See my comment in RelBuilder, which could help simplify this code.

Reply:
Good advice.